File: Java/SpaceVIL/spacevil/src/main/java/com/spacevil/CustomShape.java
package com.spacevil;

import java.util.List;

public class CustomShape extends Primitive {
    static int count = 0;
    public int RotationAngle = 32;

    /**
     * Constructs a CustomShape
     */
    public CustomShape() {
        setItemName("CustomShape_" + count);
        count++;
    }

    /**
     * Constructs a CustomShape with trinagles list
     */
    public CustomShape(List<float[]> shape) {
        this();
        setTriangles(shape);
    }

    /**
     * Make shape with triangles and convert to GL coordinates
     */
    @Override
    public List<float[]> makeShape() {
        if (getTriangles() == null || getTriangles().size() == 0)
            return null;
        return GraphicsMathService.toGL(updateShape(), getHandler());
    }
}

File: CSharp_docOnly/Items/CustomShape.cs
using System;
using System.Drawing;
using System.Collections.Generic;
using System.Linq;

namespace SpaceVIL
{
    public class CustomShape : Primitive
    {
        static int count = 0;
        /// <summary>
        /// Constructs a CustomShape
        /// </summary>
        public CustomShape()
            : base(name: "CustomShape_" + count)
        {
            count++;
        }

        /// <summary>
        /// Constructs a CustomShape with trinagles list
        /// </summary>
        public CustomShape(List<float[]> shape) : this()
        {
            SetTriangles(shape);
        }

        /// <summary>
        /// Make shape with triangles and convert to GL coordinates
        /// </summary>
        public override List<float[]> MakeShape()
        {
            if (GetTriangles() == null || GetTriangles().Count == 0)
                return null;

            return GraphicsMathService.ToGL(UpdateShape(), GetHandler());
        }
    }
}
